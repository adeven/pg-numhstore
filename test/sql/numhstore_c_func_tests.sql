SELECT array_count(ARRAY['de','de','us']);
SELECT array_count(ARRAY[1,5,9,5,10,5,5]);
SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_add('a=>1,b=>1000000000'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_add('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_sub('a=>2,b=>3'::hstore,'b=>1,c=>2'::inthstore);
SELECT array_count(ARRAY[NULL,5,NULL,5,10,NULL,5]);
SELECT array_count(ARRAY[-5,3,5]);