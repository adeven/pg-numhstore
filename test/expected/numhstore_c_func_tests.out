SELECT array_count(ARRAY['de','de','us']);
     array_count      
----------------------
 "de"=>"2", "us"=>"1"
(1 row)

SELECT array_count(ARRAY[1,5,9,5,10,5,5]);
               array_count               
-----------------------------------------
 "1"=>"1", "5"=>"4", "9"=>"1", "10"=>"1"
(1 row)

SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
          hstore_add          
------------------------------
 "a"=>"1", "b"=>"3", "c"=>"2"
(1 row)

SELECT hstore_add('a=>1,b=>1000000000'::hstore,'b=>1,c=>2'::inthstore);
              hstore_add               
---------------------------------------
 "a"=>"1", "b"=>"1000000001", "c"=>"2"
(1 row)

SELECT hstore_add('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
             hstore_add             
------------------------------------
 "a"=>"1.3", "b"=>"3.7", "c"=>"2.4"
(1 row)

SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
          hstore_add          
------------------------------
 "a"=>"1", "b"=>"3", "c"=>"2"
(1 row)

SELECT hstore_sub('a=>2,b=>3'::hstore,'b=>1,c=>2'::inthstore);
          hstore_sub           
-------------------------------
 "a"=>"2", "b"=>"2", "c"=>"-2"
(1 row)

SELECT array_count(ARRAY[NULL,5,NULL,5,10,NULL,5]);
     array_count     
---------------------
 "5"=>"3", "10"=>"1"
(1 row)

SELECT array_count(ARRAY[-5,3,5]);
ERROR:  negative integers are not supported
